api/blueprints/drbCitation.py

```diff
@@ -57,8 +58,710 @@ def citationFetch(uuid):
+                                    {citationWork.editions[0].publication_date.year}."
+            else:
+                
+                citationResponse = f"{authName}, et al. {citationWork.title}. \
+                                    translated by {transFirstName} {transLastName}, \
+                                    edited by {editorFirstName} {editorLastName}, \
+                                    {citationWork.editions[0].publishers[0]['name']}, \
+                                    {citationWork.editions[0].publication_date.year}."
+
+            return ' '.join(citationResponse.split())
+                    
+        if transName:
+            listTransName = transName.split(', ')
+            transLastName = listTransName[0]
+            transFirstName = listTransName[1]
+            authName = citationWork.authors[0]['name']
+
+            if citationWork.editions[0].publication_date < date(1900, 1, 1):
+
+                citationResponse = f"{authName}, et al. {citationWork.title}. \
+                                    translated by {transFirstName} {transLastName}, \
+                                    {citationWork.editions[0].publication_place}, \
+                                    {citationWork.editions[0].publication_date.year}."
+            else:
+
+                citationResponse = f"{authName}, et al. {citationWork.title}. \
+                                    translated by {transFirstName} {transLastName}, \
+                                    {citationWork.editions[0].publishers[0]['name']}, \
+                                    {citationWork.editions[0].publication_date.year}."
+
+            return ' '.join(citationResponse.split())
+
+        elif editorName:
+            listEditorName = editorName.split(', ')
+            editorLastName = listEditorName[0]
+            editorFirstName = listEditorName[1]
+            authName = citationWork.authors[0]['name']
+
+            if citationWork.editions[0].publication_date < date(1900, 1, 1):
+                
+                citationResponse = f"{authName}, et al. {citationWork.title}, \
+                                    edited by {editorFirstName} {editorLastName}, \
+                                    {citationWork.editions[0].publication_place}, \
+                                    {citationWork.editions[0].publication_date.year}."
+            else:
+
+                citationResponse = f"{authName}, et al. {citationWork.title}, \
+                                    edited by {editorFirstName} {editorLastName}, \
+                                    {citationWork.editions[0].publishers[0]['name']}, \
+                                    {citationWork.editions[0].publication_date.year}."
+
+            return ' '.join(citationResponse.split())
+        else:
+            mlaThreeAuthorsExtraRulesGenerator(citationWork)
+    else:
+        mlaThreeAuthorsExtraRulesGenerator(citationWork)
```

Comment:
@mitri-slory The test is failing because these two method calls are missing their `return`s!